### PR TITLE
feat (tests): Adding initial set of tests for REST

### LINKED ISSUE #357: Add test cases for services
Add test cases for services and/or pipelines components
Increase coverage requirements in last set of test in  `primeqa-ci `from 15 to 18 at least

### PR DESCRIPTION
**Pull request by kpfadnis**
# Pull Request      

## What does this PR do?

Closes #357     

## Description

* Added initial test cases for REST APIs. 
* Skipped heavier tests due to GPU constraints.
* Additional mocking may exists to circumvent running heavier models.

### CODE REVIEW

**Review by bhavani105 — APPROVED** [review]
LGTM

**Comment by bhavani105** [comment]
Please bump the version

File: tests/primeqa/services/rest_server/test_answers.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+
+@pytest.mark.skip(reason="Skipping due to gpu memory constaints ...")
```
Comment (franzmpub) [inline tests/primeqa/services/rest_server/test_answers.py:21]: "constraints" (with apologies for nitpicking)

File: tests/primeqa/services/rest_server/test_answers.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    "Canberra is the capital city of Australia. Founded following the federation of the colonies of Australia as the seat of government for the new nation, it is Australia's largest inland city"
+                ]
+            ],
+        },
+    )
+    assert response.status_code == 201
+    answers = response.json()
+
+
+@pytest.mark.skip(reason="Skipping due to gpu memory constaints ...")
```
Comment (franzmpub) [inline tests/primeqa/services/rest_server/test_answers.py:39]: "constraints"

File: tests/primeqa/services/rest_server/test_indexers.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def test_get_indexers(mock_client):
+    response = mock_client.get(
+        "/indexers",
+    )
+    assert response.status_code == 200
+    indexers = response.json()
+    assert len(indexers) == 2
+    assert ["ColBERTIndexer", "BM25Indexer"] == [
```
Comment (franzmpub) [inline tests/primeqa/services/rest_server/test_indexers.py:26]: Also "DPRIndexer"?  Or is it for later?

**Review by franzmpub — COMMENTED** [review]
Some really minor comments added.

**Review by hrojek — APPROVED** [review]
LGTM

**Review by jsmcibm — APPROVED** [review]
LGTM.  But I am surprised that there are a lot of pieces of the rest server that don't have corresponding pieces in the grpc server.  Maybe this should be elaborated in the readme.md

**Pull request merged**